Sources/ComposableArchitecture/SwiftUI/ActionSheet.swift

```diff
@@ -163,16 +163,11 @@ extension View {
     dismiss: Action
   ) -> some View {
 
-    let viewStore = ViewStore(store, removeDuplicates: { ($0 == nil) != ($1 == nil) })
```

Comment:
This was the primary issue. Replacing alert state outright would consider the change a duplicate.